libs/ui-toolkit/src/components/toast/toast.tsx

```diff
@@ -0,0 +1,165 @@
+
+import type { IconName } from '@blueprintjs/icons';
+import { IconNames } from '@blueprintjs/icons';
+import classNames from 'classnames';
+import type { ReactNode } from 'react';
+import { useEffect } from 'react';
+import { useCallback } from 'react';
+import { useLayoutEffect } from 'react';
+import { useRef } from 'react';
+import { Intent } from '../../utils/intent';
+import { Icon } from '../icon';
+
+type ToastState = 'initial' | 'showing' | 'expired';
+
+export type Toast = {
+  id: string;
+  intent: Intent;
+  children: ReactNode;
+  closeAfter?: number;
+  closeAfterAction?: number;
+};
+
+type ToastProps = Toast & {
+  state?: ToastState;
+  onClose?: (id: string) => void;
+  withStrip?: boolean;
+};
+
+const toastIconMapping: { [i in Intent]: IconName } = {
+  [Intent.None]: IconNames.HELP,
+  [Intent.Primary]: IconNames.INFO_SIGN,
+  [Intent.Success]: IconNames.TICK_CIRCLE,
+  [Intent.Warning]: IconNames.ERROR,
+  [Intent.Danger]: IconNames.ERROR,
+};
+
+const getToastAccent = (intent: Intent, withStrip = true) => ({
+  // strip
+  'bg-gray-200 text-black text-opacity-70': intent === Intent.None && withStrip,
+  'bg-vega-blue text-white text-opacity-70':
+    intent === Intent.Primary && withStrip,
+  'bg-success text-white text-opacity-70':
+    intent === Intent.Success && withStrip,
+  'bg-warning text-white text-opacity-70':
+    intent === Intent.Warning && withStrip,
+  'bg-vega-pink text-white text-opacity-70':
+    intent === Intent.Danger && withStrip,
+  // no strip
+  'text-black text-opacity-70': intent === Intent.None && !withStrip,
+  'text-vega-blue': intent === Intent.Primary && !withStrip,
+  'text-success': intent === Intent.Success && !withStrip,
+  'text-warning': intent === Intent.Warning && !withStrip,
+  'text-vega-pink': intent === Intent.Danger && !withStrip,
+});
+
+const CLOSE_DELAY = 750;
+
+export const Toast = ({
+  id,
+  intent,
+  children,
+  closeAfter,
+  closeAfterAction,
+  state = 'initial',
+  onClose,
+  withStrip = true,
+}: ToastProps) => {
+  const toastRef = useRef<HTMLDivElement>(null);
+
+  const closeToast = useCallback(() => {
+    requestAnimationFrame(() => {
+      if (toastRef.current?.classList.contains('toast-showing')) {
+        toastRef.current.classList.remove('toast-showing');
+        toastRef.current.classList.add('toast-expired');
+      }
+    });
+    setTimeout(() => {
+      onClose?.(id);
+    }, CLOSE_DELAY);
+  }, [id, onClose]);
+
+  useLayoutEffect(() => {
+    const req = requestAnimationFrame(() => {
+      if (toastRef.current?.classList.contains('toast-initial')) {
+        toastRef.current.classList.remove('toast-initial');
+        toastRef.current.classList.add('toast-showing');
+      }
+    });
+    return () => cancelAnimationFrame(req);
+  }, [id]);
+
+  useEffect(() => {
+    const onClick = (e: MouseEvent) => {
+      if (
+        toastRef.current &&
+        closeAfterAction &&
+        (e.target instanceof HTMLButtonElement ||
```

Comment:
suggestion (non blocking): What if there are multiple buttons/anchors to do different things within the toast content? I guess that might unlikely.. but at a glance it might be better to make `children` a render prop and pass the toast id down to the toast content. Then the toast content can use the toast store to close itself on click of whatever element it wants.

This would solve that problem and keep us a little more in the react paradigm